flink-core/src/main/java/org/apache/flink/types/Row.java

```diff
@@ -45,6 +45,9 @@
  *
  * <p>A row instance is in principle {@link Serializable}. However, it may contain non-serializable fields
  * in which case serialization will fail if the row is not serialized with Flink's serialization stack.
+ *
+ * <p>The {@link #equals(Object)} and {@link #hashCode()} methods of this class support all conversion
```

Comment:
I think you mean "all **external** conversion classes"? Because it doesn't support the internal conversion classes, e.g. `MapData`.

Reply:
We fixed MapData in FLINK-17668. I think most internal data structures have a proper hashCode/equals now. But yes, we can also be more specific here.